include/seqan3/alignment/matrix/alignment_trace_matrix.hpp

```diff
@@ -98,75 +99,148 @@ struct alignment_coordinate
     size_t seq2_pos;
 };
 
-/*!\brief Compute the trace from a trace matrix
+/*!\brief Compute the begin coordinate.
  * \ingroup alignment_matrix
- * \tparam    database_t     The type of the database sequence.
- * \tparam    query_t        The type of the query sequence.
  * \tparam    trace_matrix_t The type of the trace matrix.
- * \param[in] database       The database sequence.
- * \param[in] query          The query sequence.
  * \param[in] matrix         The trace matrix.
  * \param[in] end_coordinate Where the trace in the matrix ends.
  */
+ template <typename trace_matrix_t>
+ //!\cond
+     requires matrix_concept<remove_cvref_t<trace_matrix_t>> &&
+              std::Same<typename remove_cvref_t<trace_matrix_t>::entry_type, trace_directions>
+ //!\endcond
+inline alignment_coordinate
+alignment_begin_coordinate(trace_matrix_t && matrix, alignment_coordinate const end_coordinate)
+{
+    using signed_size_t = std::make_signed_t<size_t>;
+
+    constexpr auto N = trace_directions::none;
+    constexpr auto D = trace_directions::diagonal;
+    constexpr auto L = trace_directions::left;
+    constexpr auto U = trace_directions::up;
+    signed_size_t row = end_coordinate.seq2_pos + 1;
+    signed_size_t col = end_coordinate.seq1_pos + 1;
+
+    assert(row < matrix.rows());
+    assert(col < matrix.cols());
+
+    while(true)
+    {
+        trace_directions dir = matrix.at(row, col);
+        if ((dir & L) == L)
+        {
+            col = std::max<signed_size_t>(col - 1, 0);
+        }
+        else if ((dir & U) == U)
+        {
+            row = std::max<signed_size_t>(row - 1, 0);
+        }
+        else if ((dir & D) == D)
+        {
+            row = std::max<signed_size_t>(row - 1, 0);
+            col = std::max<signed_size_t>(col - 1, 0);
+        }
+        else
+        {
+            if (row == 0 || col == 0)
+                break;
+            throw std::logic_error{"Trace not found"};
+        }
+    }
+    return
+    {
+        std::max<signed_size_t>(col - 1, 0),
+        std::max<signed_size_t>(row - 1, 0)
+    };
```

Comment:
please introduce a blank line before the return statement and write the return statement into a single line, it looks really weird :see_no_evil:

Reply:
yeah readability looks weird...